### PR TITLE
Math magicians: events

### PR DESCRIPTION
**Pull request by zemola**
- Copy the files calculate.js and operate.js into a logic/ directory of my project.
- Analyze the files calculate.js and operate.js, to understand what they do.
- Import the files in my Calculator component.
- Implement the event handlers I need to use the math logic from calculate.js and operate.js in my React component.
- used state and props, pass props to child components, use lifecycle methods, lift state up.
- Tested my app in the browser. and It's able to perform math operations.

### CODE REVIEW

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
```markdown
Hi Azeem, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not follow Git flow. Plesae create a `development` branch, make it a default branch and open PR from feature to development branch. 
Furthermore, Please add `npm install` and `npm start` commands in README so that a user is able to run the project locally.
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

Invalid Code Review Request **does not count** into the code reviews limit._
```

File: src/componets/Calculator.js [hunk 1]
```diff
@@ -1,43 +1,66 @@
...
+          <Btn value="4" className="btn" click={handleClick} />
+          <Btn value="5" className="btn" click={handleClick} />
+          <Btn value="6" className="btn" click={handleClick} />
+          <Btn value="-" className="btn light" click={handleClick} />
+        </div>
+        <div className="btn-wrapper">
+          <Btn value="1" className="btn" click={handleClick} />
+          <Btn value="2" className="btn" click={handleClick} />
+          <Btn value="3" className="btn" click={handleClick} />
+          <Btn value="+" className="btn light" click={handleClick} />
```
Comment (D4nielJ) [inline src/componets/Calculator.js]: - [ ] Currently, there are two `+` plus signs in the calculator and you're missing the `+/-` plus / less sign. Let's fix this as this is part of the original design of the calculator.

File: src/componets/Calculator.js [hunk 2]
```diff
@@ -1,43 +1,66 @@
 import React from 'react';
+import Btn from './button';
 import './calculator.css';
+import calculate from '../logic/calculate';
 
 // eslint-disable-next-line react/prefer-stateless-function
```
Comment (D4nielJ) [inline src/componets/Calculator.js]: [OPTIONAL] I think you can delete this disable linter rule now that you're using a component with state.

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @zemola, 

You're doing great! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Deleting the `app.test.js` as it is part of the boilerplate created from CRA and you're not using it now and it causes the test to fail.

image


## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**